files/nrpe/check_openstack_usage_stats.py

```diff
@@ -23,7 +23,10 @@
 NAGIOS_STATE_UNKNOWN  = 3
 
 def get_all_servers(nova):
-    return nova.servers.list(search_opts={'all_tenants': True})
+    search_opts = {'all_tenants': True,
+                   'deleted': False,
+                   'instance_name': False}
```

Comment:
I couldn't find Newton API reference only latest [1]. Nor could I find it while grepping in nova's api.py (near same place where I could find for example access_ip_v4). So I tried the code without the "instance_name: False" and it was still fast.
Perhaps that bit could be removed if you agree it's not used anyway?

[1]: https://docs.openstack.org/api-ref/compute/?expanded=list-servers-detailed-detail#list-servers

Reply:
I removed all the get `get_all_servers()` changes since they were unnecessary, I had just copied them from the `get_per_flavor_active_vm_count()` since I thought they were necessary. The best thing of course when I'm doing a git blame to see who have created the unnecessary search_opts it is my self.